client/state/jetpack-connect/selectors.js

```diff
@@ -103,33 +102,29 @@ const getUserAlreadyConnected = state => {
 };
 
 /**
- * XMLRPC errors can be identified by the presence of an error message, the presence of an authorization code
- * and if the error message contains the string 'error'
+ * Returns true if there is an XMLRPC error.
  *
- * @param {object} state Global state tree
- * @returns {Boolean} If there's an xmlrpc error or not
+ * @param  {object}  state Global state tree
+ * @return {Boolean}       True if there's an xmlrpc error or not
  */
 const hasXmlrpcError = function( state ) {
 	const authorizeData = getAuthorizationData( state );
 
-	return (
-		authorizeData &&
-		authorizeData.authorizeError &&
-		authorizeData.authorizeError.message &&
-		authorizeData.authorizationCode &&
```

Comment:
Relevant?

Reply:
I believe `authorizationCode` is still relevant, unless we want this selector to catch other non-XMLRPC errors here (like error with access code generation for example). And I think the addition of `authorizationCode` here was with the sole purpose to exclude non-XMLRPC errors.